src/components/item/ItemsContainer.js

```diff
@@ -8,133 +8,30 @@ import Item from 'components/item/Item';
 const ItemsContainer = () => {
     const { state, dispatch } = useStore();
     const {
-      size,
       cart,
       products,
       hasBeenFiltered,
       currentProducts
     } = state;
 
-    // Add selected item to customers cart
-    const addToCart = (item, numberOfItems) => {
-      // Check if customer selected the size 
-      if (size.size === null || size.id !== item) {
```

Comment:
just declare `formErrors.size = !state.size`, don't check here and invoke some side-effect